rdflib/graph.py

```diff
@@ -1944,18 +1945,19 @@ class Dataset(ConjunctiveGraph):
     """
 
     def __init__(self, store="default", default_union=False, default_graph_base=None):
-        super(Dataset, self).__init__(store=store, identifier=None)
+        super(Dataset, self).__init__(store=store, identifier=DATASET_DEFAULT_GRAPH_ID)
 
         if not self.store.graph_aware:
             raise Exception("DataSet must be backed by a graph-aware store!")
-        self.default_context = Graph(
-            store=self.store,
-            identifier=DATASET_DEFAULT_GRAPH_ID,
-            base=default_graph_base,
-        )
 
         self.default_union = default_union
 
+    def add(self, triple_or_quad):
```

Comment:
I think having `add` return a reference to the graph is dumb, but for consistency with the other `Graph` implementations, it should return the return value of `add` in this method.